doc/source/providers/amazon.rst

```diff
@@ -149,5 +149,457 @@ On completion, you should receive an ID for your transaction.
+    INFO:                   Instance Info
+    INFO: --------------------------------------------------
+    INFO:  * ID: 2189
+    INFO:  * NAME: aws_vm_1
+    INFO: --------------------------------------------------
+
+2. Delete/Terminate the Instance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+    :linenos:
+
+    (miq-client) $ miqcli instances terminate
+
+You can view the available options by passing the **help** option.
+
+.. code-block:: bash
+    :linenos:
+
+    (miq-client) $ miqcli instances terminate --help
+
+This command when run will send a request to ManageIQ for a new transaction to
+delete the instance. Before running the command, replace the **<<value>>**
+with your Amazon virtual machine details. For <instance_name> provide the ID returned
+by the query command above and set the option **by_id** to True.
+
+.. code-block:: bash
+    :linenos:
+
+    (miq-client) $ miqcli instances terminate \
+    <instance_name>
+    --by_id True
+
+On completion, you should receive an ID for your transaction.
+
+.. code-block:: bash
+
+    INFO: --------------------------------------------------
+    INFO:                   Instance Info
+    INFO: --------------------------------------------------
+    INFO:  * ID: 2189
+    INFO:  * NAME: aws_vm_1
+    INFO: --------------------------------------------------
+    INFO: Task to terminate 2189 created: 1324
+
+Use 'miqcli tasks status <ID>' to check status of transaction completion.
+
+3. Remove the VM reference in Cloudforms/MIQ
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+    :linenos:
+
+    (miq-client) $ miqcli vms delete
+
+You can view the available options by passing the **help** option.
+
+.. code-block:: bash
+    :linenos:
+
+    (miq-client) $ miqcli vms delete --help
+
+This command when run will send a request to ManageIQ for a new transaction to
+delete the vms reference. Before running the command, replace the **<<value>>**
+with your Amazon virtual machine details. For <instance_name> provide the ID returned
+by the query command above and set the option **by_id** to True.
+
+.. code-block:: bash
+    :linenos:
+
+    (miq-client) $ miqcli vms delete \
+    <instance_name>
+    --by_id True
+
+On completion, you should receive an ID for your transaction.
+
+.. code-block:: bash
+
+    INFO: --------------------------------------------------
+    INFO:                   Vm Info
+    INFO: --------------------------------------------------
+    INFO:  * ID: 2189
+    INFO:  * NAME: aws_vm_1
+    INFO: --------------------------------------------------
+    INFO: Task to delete 2189 created: 1324
+
+Use 'miqcli tasks status <ID>' to check status of transaction completion.
+
+Python Code
++++++++++++
+
+This code example will demonstrate how you can use the client in a Python
+module to delete a virtual machine. Before running the Python code, replace the
+**<value>** with your Amazon virtual machine details.
+
+.. code-block:: python
```

Comment:
Instead of adding the python code block which looks to be a mirror of the `manageiq-cli/examples/aws_deletion_example.py`. We could use [literalinclude](http://www.sphinx-doc.org/en/stable/markup/code.html#directive-literalinclude). to include the source code directly in the docs. This way updates to the source code are reflected to the docs pages. What do you think?